src/modules/Builder/pages/BuilderApplet/BuilderApplet.tsx

```diff
@@ -61,17 +64,21 @@ export const BuilderApplet = () => {
   } = methods;
 
   useEffect(() => {
-    if (loadingStatus === 'success' && !isNewApplet) {
+    if (location.state?.isFromLibrary) setIsFromLibrary(true);
+  }, [location.state]);
+
+  useEffect(() => {
+    if (
+      loadingStatus === 'success' &&
```

Comment:
pls use a variable for bunch of conditions and use early return to avoid nesting

Reply:
fixed